File: VB/Scaffolding.DetailCollections/ViewModels/DepartmentViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports Scaffolding.DetailCollections.Common.Utils
Imports Scaffolding.DetailCollections.DepartmentContextDataModel
Imports Scaffolding.DetailCollections.Common.DataModel
Imports Scaffolding.DetailCollections.Model
Imports Scaffolding.DetailCollections.Common.ViewModel

Namespace Scaffolding.DetailCollections.ViewModels
	Partial Public Class DepartmentViewModel
		Inherits SingleObjectViewModel(Of Department, Integer, IDepartmentContextUnitOfWork)
		Public Sub New()
			Me.New(UnitOfWorkSource.GetUnitOfWorkFactory())
		End Sub
		Public Sub New(ByVal unitOfWorkFactory As IUnitOfWorkFactory(Of IDepartmentContextUnitOfWork))
			MyBase.New(unitOfWorkFactory, Function(x) x.Departments)
		End Sub
		Protected Overrides Sub RefreshLookUpCollections(ByVal key As Integer)
			DepartmentCoursesLookUp = CreateLookUpCollectionViewModel(Function(x) x.Courses, Function(x) x.DepartmentID = key, key)
			DepartmentEmployeesLookUp = CreateLookUpCollectionViewModel(Function(x) x.Employees, Function(x) x.DepartmentID = key, key)
		End Sub
		Private privateDepartmentCoursesLookUp As LookUpCollectionViewModel(Of Department, Course, Integer, IDepartmentContextUnitOfWork)
		Public Overridable Property DepartmentCoursesLookUp() As LookUpCollectionViewModel(Of Department, Course, Integer, IDepartmentContextUnitOfWork)
			Get
				Return privateDepartmentCoursesLookUp
			End Get
			Set(ByVal value As LookUpCollectionViewModel(Of Department, Course, Integer, IDepartmentContextUnitOfWork))
				privateDepartmentCoursesLookUp = value
			End Set
		End Property
		Private privateDepartmentEmployeesLookUp As LookUpCollectionViewModel(Of Department, Employee, Integer, IDepartmentContextUnitOfWork)
		Public Overridable Property DepartmentEmployeesLookUp() As LookUpCollectionViewModel(Of Department, Employee, Integer, IDepartmentContextUnitOfWork)
			Get
				Return privateDepartmentEmployeesLookUp
			End Get
			Set(ByVal value As LookUpCollectionViewModel(Of Department, Employee, Integer, IDepartmentContextUnitOfWork))
				privateDepartmentEmployeesLookUp = value
			End Set
		End Property
	End Class
End Namespace

File: CS/Scaffolding.DetailCollections/ViewModels/DepartmentViewModel.cs
using System;
using System.Linq;
using System.Linq.Expressions;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using Scaffolding.DetailCollections.Common.Utils;
using Scaffolding.DetailCollections.DepartmentContextDataModel;
using Scaffolding.DetailCollections.Common.DataModel;
using Scaffolding.DetailCollections.Model;
using Scaffolding.DetailCollections.Common.ViewModel;

namespace Scaffolding.DetailCollections.ViewModels {
    public partial class DepartmentViewModel : SingleObjectViewModel<Department, int, IDepartmentContextUnitOfWork> {
        public DepartmentViewModel()
            : this(UnitOfWorkSource.GetUnitOfWorkFactory()) {
        }
        public DepartmentViewModel(IUnitOfWorkFactory<IDepartmentContextUnitOfWork> unitOfWorkFactory)
            : base(unitOfWorkFactory, x => x.Departments) {
        }
        protected override void RefreshLookUpCollections(int key) {
            DepartmentCoursesLookUp = CreateLookUpCollectionViewModel(x => x.Courses, x => x.DepartmentID == key, key);
            DepartmentEmployeesLookUp = CreateLookUpCollectionViewModel(x => x.Employees, x => x.DepartmentID == key, key);
        }
        public virtual LookUpCollectionViewModel<Department, Course, int, IDepartmentContextUnitOfWork> DepartmentCoursesLookUp { get; set; }
        public virtual LookUpCollectionViewModel<Department, Employee, int, IDepartmentContextUnitOfWork> DepartmentEmployeesLookUp { get; set; }
    }
}
